src/Mvc/MvcTemplates/N2/Files/FileSystem/Directory.aspx.cs

```diff
@@ -297,6 +299,17 @@ protected string ImageBackgroundStyle(string url)
 			}
 		}
 
+        protected string GetEditUrl()
+        {
+            if (Selection.SelectedItem == null) throw new ArgumentNullException("selected");
 
-	}
+            ContentItem parent = Selection.SelectedItem;
+
+            Url url = Url.ResolveTokens("{ManagementUrl}/Content/Edit.aspx");

```

Comment:
This is great! Were you able to find this being used elsewhere? It's nice that you were able to find ways to make this portion dynamic for the most part.